pkg/controller/controller.go

```diff
@@ -440,17 +442,18 @@ func (c *Controller) handleClusterUpdate(cluster *myspec.M3DBCluster) error {
 
 	c.logger.Info("found placement", zap.Int("currentPods", len(pods)), zap.Int("placementInsts", placement.NumInstances()))
 
-	if placement.NumInstances() < len(pods) {
-		for _, pod := range pods {
-			_, ok := placement.Instance(pod.Name)
-			if !ok {
-				// We want pod adds to restart the event loop to run regular sanity
-				// checks.
-				return c.addPodToPlacement(cluster, pod, placement)
-			}
+	unavailInsts := []string{}
+	for _, inst := range placement.Instances() {
+		if !inst.IsAvailable() {
+			unavailInsts = append(unavailInsts, inst.ID())
 		}
 	}
 
+	if len(unavailInsts) > 0 {
+		c.logger.Info("waiting for instances to be available", zap.Strings("instances", unavailInsts))
+		return nil
```

Comment:
Is there any potential that non-scaling events will be dropped if the cluster receives updates for other aspects?

Reply:
The one event that I am worried about are deployments. It would require the human operator to look at the logs to understand why the deployment isn't running but that is reasonable. Could we change this to a Warn log message so that it sticks out like a sore thumb?

Reply:
Yeah definitely, good point